### DISCUSSION TITLE
couldnt reach local plex server at server address: http://localhost:32400 - or this library really is empty

### DISCUSSION THREAD

**Discussion by I-am-PUID-0** [itsToggle/plex_debrid#205] [General]
Hello All,

plex_debrid has started throwing an error when attempting to run. 

Note: Plex_debrid was previously running with the current settings/configuration. I also noted the other similar discussion; however, the issue is unrelated as /web was appended to the plex server address in that discussion.

Configuration:
-OS: Server 2012 R2
-Python 3.9.1
-Plex Version 1.29.0.6244
-Both Plex and plex_debrid are running from the same host on the host OS (not virtualized) 
-"Plex server address": "http://localhost:32400"
-"Plex \"movies\" library": "2",
-"Plex \"shows\" library": "4",

plex_debrid is throwing the following error:
-[plex] getting entire plex library ... done
-[plex error] couldnt reach local plex server at server address: http://localhost:32400 - or this library really is empty

Plex logs show connection attempts from 127.0.0.1:
-Request: [127.0.0.1:49472 (Loopback)] GET /library/all (12 live) #bd9f4 GZIP Signed-in Token (username removed) (Home Server)
-[Req#1be0eb] It took 3984.375000 ms to retrieve 19657 items.
-Completed after connection close: [127.0.0.1:49472] 200 GET /library/all (10 live) GZIP 37041ms 65536 bytes (pipelined: 1)

Troubleshooting:
-Reboot (yes, I turned it off then on)
-Clean download/install
-Removed setting.json and setup
-Loaded from VM and adjusted plex server address to IP
-Downgrade of Plex server
-Upgrade of Plex server 
-Access Plex at localhost:32400

Any help is very much appreciated. THX!

**Comment by itsToggle** [comment]
Now this is how debugging should be done! Thanks for your detailed description! :)

Inside plex_debrid, could you go to /Settings/UI Settings/debug printing/ and set it to "true"? Perhaps there is a more detailed log available, that could give us a hint to what's going wrong. 

What version of plex_debrid are you running? The settings 'Plex "movies" library' and 'Plex "shows" library' are deprecated since V2.10 - but it sounds like you did do a clean download/reinstall, so I'm assuming you are up to date.

Other than that I would have recommended exactly what you already did. The request to plex did take a bit of time, but the plex_requests are set to time out after 30 seconds, so that shouldn't be an issue.

**Reply by I-am-PUID-0** [reply]
@itsToggle, thanks for the reply! 

Apologies; I should have included the plex_debrid versions tested. The issue is present with versions 2.00 and 2.15.

The debug print provided the following additional details: 
-plex error: <json exception>: HTTPConnectionPool<host=' localhost' , port=32400>: Read timed out. read timeout=30

Based on the additional detail of the debug print, I increased the timeout to 60 in the content/services/plex.py file. 

It's now up and running again!!

Thanks for the insight and the great project!

**Reply by itsToggle** [reply]
Sure thing! I will increase the timeout in the next version :) By default, plex_debrid will scan all your plex libraries, including plex "shares", for existing content. If you have any libraries that you want to exclude from the scan (Plex shares perhaps), you could do so by navigating to /Settings/Content Services/Plex/Plex library scan/.
Glad its working now though!

**Reply by I-am-PUID-0** [reply]
@itsToggle, should I create an issue from this discussion so that the resolution is tracked?

**Reply by itsToggle** [reply]
Hey, the timeout has already been increased to 60 seconds :)

**Reply by I-am-PUID-0** [reply]
Ah, excellent! I just ran v2.25 and the timeout does appear to be working. Thx! 

Unfortunately, v2.25 also appears to be stuck just after "checking new content"

Before enabling the debug print it shows the following:
-[plex] getting entire plex library ... done
-[plex] getting all watchlists ... done
-checking new content ... done
with no additional updates for a prolonged period. I exited after about 30 mins and enabled the debug print. 

After enabling the debug print it shows the following:
-[plex] getting entire plex library ... done
-[plex] getting all watchlists ... done
-checking new content ... done
Then it continues to print "[plex] updating all watchlists ... done" every 5 s

I let it run for an hour with no change in the state. 

All settings and configurations are as they were in the root discussion, except for the new settings.json format being utilized. 
    "Library collection service": [
        "Plex Library"
    ],
    "Library update services": [
        "Plex Libraries"
    ],
    "Library ignore services": [
        "Local Ignore List"
    ],
    "Trakt library user": [],
    "Trakt refresh user": [],
    "Plex library refresh": [
        "2",
        "4"
    ],

Thanks again!

**Reply by I-am-PUID-0** [reply]
Disregard the last issue. 

I resolved it, and I'm posting the resolution in the event someone else has the same result. 

I failed to add the Trakt Lists in Options/Settings/Content Services/Content Services/Trakt: 
or as defined in the settings.json:
"Trakt lists": [
        "username removed watchlist",
        "username removed collection"
    ],

Thx!
